HardwareReport/HardwareReport.js

```diff
@@ -1166,27 +1166,76 @@ function load_airspeed(log) {
 }
 
 // Load GPS
-let gps
+let gps;
 const max_num_gps = 2
-function load_gps() {
+let num_gps = 0
+function load_gps(log) {
+    let gps_instances = []
 
     for (let i = 0; i < max_num_gps; i++) {
-        let index = String(i+1)
+        let index = String(i + 1)
         if (i == 0) {
             index = ""
         }
         const type_name = "GPS_TYPE" + index
         if (type_name in params) {
             const type = params[type_name]
             if (type != 0) {
-                const pos_names = get_param_name_vector3("GPS_POS" + (i+1) + "_")
-                gps[i] = { type: type,
-                           pos: get_param_array(params, pos_names),
-                           node_id: params["GPS_CAN_NODEID" + (i+1)] }
+                const pos_names = get_param_name_vector3("GPS_POS" + (i + 1) + "_")
+                gps[i] = {
+                    type: type,
+                    pos: get_param_array(params, pos_names),
+                    node_id: params["GPS_CAN_NODEID" + (i + 1)]
+                }
             }
         }
     }
 
+    let messages = log.get("MSG").Message
+
+    let section = document.getElementById("GPS")
+    let gps_num = 0
+    // This regular expression is used to get the word after "as" to get the GPS device name. (?<=as\s) will exclude the "as" and a whitespace from the regex and then (\S+) will match the next word.
+    let regex = /(?<=as\s)(\S+)/i;
+    for (let i = 0; i < messages.length; i++) {
+        if (messages[i].startsWith("GPS")) {
+            let gps_device = messages[i].match(regex)[0];
+            gps_instances[gps_num++] = { gps: gps_device }
```

Comment:
I think you should be able to pull the number out of the message too? This will mean it still does the right thing if we get two detection messages for the same GPS for some reason. Or if we miss the one for the first gps and only get the second.